index.bs

```diff
@@ -5483,6 +5497,267 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+The [=remote end steps=] are:
+
+ 1. If |parameters| is not a JSON [=Object=], return a [=WebDriver error=] with [=WebDriver error code=]
+     [=invalid argument=].
+ 1. Let |credentialId| be the result of [=trying=] to decode [=Base64url Encoding=] on the |parameter|'s
+     {{CredentialParams/credentialId}} property.
+ 1. If |credentialId| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |rpIdHash| be the result of [=trying=] to decode [=Base64url Encoding=] on the |parameter|'s
+     {{CredentialParams/rpIdHash}} property.
+ 1. If |rpIdHash| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. If |rpIdHash| is not a valid [=rpIdHash=], return a [=WebDriver error=] with [=WebDriver error code=]
+    [=invalid argument=].
+ 1. Let |privateKey| be the result of [=trying=] to decode [=Base64url Encoding=] on the |parameter|'s
+     {{CredentialParams/privateKey}} property.
+ 1. If |privateKey| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. If |privateKey| is not a valid PKCS #8 encoded private key [[RFC5958]], return a [=WebDriver error=] with
+     [=WebDriver error code=] [=invalid argument=].
+ 1. Create a [=Credential Key Pair=] from |privateKey| identified by |credentialId| and an [=Authenticator Data=]
+     structure from |rpIdHash| and |signCount|, and store them in the database of the [=Virtual Authenticator=]
+     identified by |authenticatorId|.
+ 1. Return [=success=].
+
+## <dfn>Get Credentials</dfn> ## {#sctn-automation-get-credentials}
+
+The [=Get Credentials=] [=extension command=] returns one {{CredentialParams}} for every [=Credential Key Pair=]
+and associated [=Authenticator Data=] stored in a [=Virtual Authenticator=], regardless of whether they were
+stored using [=Add Credential=] or {{CredentialsContainer/create()|navigator.credentials.create()}}. The [=extension command=]
+is defined as follows:
+
+<figure id="table-getCredentials" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>GET</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credentials`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. For every [=Credential Key Pair=], construct a corresponding {{CredentialParams}}.
```

Comment:
should this be something more like:

Let |credentialsArray| be an empty array.
For each public key credential source (|credential|) managed by the authenticator identified by |authenticatorId|, construct a corresponding {{CredentialParams}} object and add it to |credentialsArray|.

..?

Reply:
Done.